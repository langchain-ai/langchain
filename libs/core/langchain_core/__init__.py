"""``langchain-core`` defines the base abstractions for the LangChain ecosystem.

The interfaces for core components like chat models, LLMs, vector stores, retrievers,
and more are defined here. The universal invocation protocol (Runnables) along with
a syntax for combining components (LangChain Expression Language) are also defined here.

No third-party integrations are defined here. The dependencies are kept purposefully
very lightweight.
"""

from langchain_core._api import (
    surface_langchain_beta_warnings,
    surface_langchain_deprecation_warnings,
)
from langchain_core.version import VERSION

__version__ = VERSION

surface_langchain_deprecation_warnings()
surface_langchain_beta_warnings()
