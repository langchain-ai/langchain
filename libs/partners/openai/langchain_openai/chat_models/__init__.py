from langchain_openai.chat_models.azure import AzureChatOpenAI
from langchain_openai.chat_models.base import ChatOpenAI

__all__ = ["ChatOpenAI", "AzureChatOpenAI"]
