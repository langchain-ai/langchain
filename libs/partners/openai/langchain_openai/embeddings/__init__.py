from langchain_openai.embeddings.azure import AzureOpenAIEmbeddings
from langchain_openai.embeddings.base import OpenAIEmbeddings

__all__ = ["OpenAIEmbeddings", "AzureOpenAIEmbeddings"]
