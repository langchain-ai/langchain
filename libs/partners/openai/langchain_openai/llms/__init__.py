from langchain_openai.llms.azure import AzureOpenAI
from langchain_openai.llms.base import OpenAI

__all__ = ["OpenAI", "AzureOpenAI"]
