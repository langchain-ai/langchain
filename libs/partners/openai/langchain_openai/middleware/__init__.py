"""Middleware implementations for OpenAI-backed agents."""

from .openai_moderation import OpenAIModerationError, OpenAIModerationMiddleware

__all__ = [
    "OpenAIModerationError",
    "OpenAIModerationMiddleware",
]
