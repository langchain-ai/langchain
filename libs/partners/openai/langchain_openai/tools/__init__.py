from langchain_openai.tools.custom_tool import custom_tool

__all__ = ["custom_tool"]
