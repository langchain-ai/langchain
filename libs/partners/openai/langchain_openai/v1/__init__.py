from langchain_openai.v1.chat_models import ChatOpenAI

__all__ = ["ChatOpenAI"]
