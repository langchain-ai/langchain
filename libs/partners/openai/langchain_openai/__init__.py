from langchain_openai.chat_models import AzureChatOpenAI, ChatOpenAI
from langchain_openai.embeddings import AzureOpenAIEmbeddings, OpenAIEmbeddings
from langchain_openai.llms import AzureOpenAI, OpenAI

__all__ = [
    "OpenAI",
    "ChatOpenAI",
    "OpenAIEmbeddings",
    "AzureOpenAI",
    "AzureChatOpenAI",
    "AzureOpenAIEmbeddings",
]
