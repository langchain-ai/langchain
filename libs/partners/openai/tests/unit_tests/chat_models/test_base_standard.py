"""Standard LangChain interface tests"""

from typing import Tuple, Type

from langchain_core.language_models import BaseChatModel
from langchain_tests.unit_tests import ChatModelUnitTests

from langchain_openai import ChatOpenAI


class TestOpenAIStandard(ChatModelUnitTests):
    @property
    def chat_model_class(self) -> Type[BaseChatModel]:
        return ChatOpenAI

    @property
    def init_from_env_params(self) -> Tuple[dict, dict, dict]:
        return (
            {
                "OPENAI_API_KEY": "api_key",
                "OPENAI_ORG_ID": "org_id",
                "OPENAI_API_BASE": "api_base",
                "OPENAI_PROXY": "https://proxy.com",
            },
            {},
            {
                "openai_api_key": "api_key",
                "openai_organization": "org_id",
                "openai_api_base": "api_base",
                "openai_proxy": "https://proxy.com",
            },
        )
