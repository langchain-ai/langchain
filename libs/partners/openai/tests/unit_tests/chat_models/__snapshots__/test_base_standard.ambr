# serializer version: 1
# name: TestOpenAIStandard.test_serdes[serialized]
  dict({
    'id': list([
      'langchain',
      'chat_models',
      'openai',
      'ChatOpenAI',
    ]),
    'kwargs': dict({
      'max_retries': 2,
      'max_tokens': 100,
      'model_name': 'gpt-3.5-turbo',
      'openai_api_key': dict({
        'id': list([
          'OPENAI_API_KEY',
        ]),
        'lc': 1,
        'type': 'secret',
      }),
      'request_timeout': 60.0,
      'stop': list([
      ]),
      'temperature': 0.0,
    }),
    'lc': 1,
    'name': 'ChatOpenAI',
    'type': 'constructor',
  })
# ---
